src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -544,7 +544,7 @@ public void CopyTo(T[] array, int arrayIndex)
 
                 if (Count > (array.Length - arrayIndex))
                 {
-                    throw new ArgumentException("arrayIndex");
+                    throw new ArgumentException(null, nameof(arrayIndex));
```

Comment:
ArgumentNullException is better.

Reply:
Again, `arrayIndex` may not be null here.

Reply:
Sorry, I meant `PSArgumentException(nameof(arrayIndex))`.

Reply:
`PSArgumentException(nameof(arrayIndex))` would be [CA2208](https://docs.microsoft.com/visualstudio/code-quality/ca2208) as method signature is `PSArgumentException(string message)`.